core/trino-main/src/main/java/io/trino/sql/planner/plan/GroupIdNode.java

```diff
@@ -103,6 +100,15 @@ public List<List<Symbol>> getGroupingSets()
         return groupingSets;
     }
 
+    @JsonIgnore
+    public Set<Symbol> getDistinctGroupingSetSymbols()
+    {
+        // Produce a stable ordering of grouping set symbols in the output
+        ImmutableSet.Builder<Symbol> builder = ImmutableSet.builder();
+        groupingSets.forEach(builder::addAll);
+        return builder.build();
```

Comment:
This can be:

```java
return groupingSets.stream()
                .flatMap(Collection::stream)
                .collect(ImmutableSet.toImmutableSet())
```

Reply:
Done